core/workspace.js

```diff
@@ -123,18 +135,6 @@ Blockly.Workspace.SCAN_ANGLE = 3;
  */
 Blockly.Workspace.prototype.addTopBlock = function(block) {
   this.topBlocks_.push(block);
-  if (!this.isFlyout) {
-    return;
-  }
-  // This is for the (unlikely) case where you have a variable in a block in
-  // an always-open flyout.  It needs to be possible to edit the block in the
-  // flyout, so the contents of the dropdown need to be correct.
-  var variableNames = Blockly.Variables.allUsedVariables(block);
-  for (var i = 0, name; name = variableNames[i]; i++) {
-    if (!this.getVariable(name)) {
-      this.createVariable(name);
-    }
-  }
```

Comment:
What is the reason for this change?

Reply:
For getting rid of this?  Because we're now creating the variable as part of initModel or in XML deserialization, and doing it in all cases instead of just in this special case.  Also because that case is obviously not unlikely if it happens on every single use of scratch-blocks.